### ISSUE TITLE
warnings, some involving ada 2022

### ISSUE THREAD

**Issue by johnperry-math** [zertovitch/gid#1]
Hi! I'm using GID in a project, and when I `alr build` with `pragma Ada_2022;` in my source files, I get some warnings:

#### Array aggregate notation
```
project.adb:65:07: warning: array aggregate using () is an obsolescent syntax, use [] instead [-gnatwj]                                                                                   
project.adb:65:07: warning: in instantiation at gid-decoding_png.adb:1126 [-gnatwj]
project.adb:65:07: warning: in instantiation at gid-decoding_png.adb:1466 [-gnatwj]
project.adb:65:07: warning: in instantiation at gid.adb:136 [-gnatwj]
```
~~It's easy to eliminate on my end by adding `pragma Ada_12 (GID);` to my code, but~~ would you consider adding `pragma Ada_2012` to these source files?

(whoops, I thought that compiled when I added it on my end)

#### Unused variables
```
gid-decoding_png.adb:743:09: warning: variable "slide" is never read and never assigned [-gnatwu]
gid-decoding_png.adb:746:09: warning: variable "crc32val" is never read and never assigned [-gnatwu]
```
I haven't looked into these, but I find it surprising that `crc32val` is unused. I don't know if you think it worth eliminating them, but it seems a good idea to report it.

I enjoy the package for now, by the way!

**Comment by zertovitch** [comment]
About the first bunch of warnings, perhaps you don't get them when you remove `pragma Ada_2022;` ?
Or does it prevent you to use the 2022 syntax elsewhere?
Another way is to remove the `-gnatwj` switch, using `-gnatwJ`.
Adding `pragma Ada_2012` ? Really not my preferred option...

Regarding the second batch, it is a GNAT bug. Try commenting out `slide` for instance...

**Comment by zertovitch** [comment]
Anyway, thanks for the report.
The GNAT erroneous warning has had a nice side effect: I have reviewed the code!
The crc32 checksum was computed but not checked, for the good reason that it should be actually an Adler32!...
Fixed in commit a4ff7ba.

**Comment by johnperry-math** [comment]
Sorry for the late follow up.

> About the first bunch of warnings, perhaps you don't get them when you remove `pragma Ada_2022;`?

That's correct. I included `pragma Ada_2012;` in the spec of the package that uses gid, changed the body code so that it no longer relied on Ada 2022 features, and the warnings about array aggregates went away. I am not otherwise able to tell Ada that, say, GID uses Ada 2012, which disappoints me.

> Another way is to remove the `-gnatwj` switch, using `-gnatwJ`.

I'll have to see if alire will let me apply that on a case-by-case basis.

> Adding `pragma Ada_2012`? Really not my preferred option...

OK. I figured it couldn't hurt to ask. :grin:

> The crc32 checksum was computed but not checked, for the good reason that it should be actually an Adler32!...

I do remember Adler32 being an essential part of PNG, but I remember CRC-32 is also a part of it. whoops!
